Week-3/a1/index.js

```diff
@@ -0,0 +1,51 @@
+/*
+memoize function will first sort the array descending order, and then calculate or retrieve sum.
+this helps in reducing calculations for shuffled but same parameters, for eg. (100, 200, 300) or (200, 100, 300) or (300, 200, 100) 
+*/
+function memoize(fn) {
+    const cache = new Map();
+    return function(...args) {
+        console.log('---------------------------------------------');
+        console.log('original array =', args.toString());
+        var sum = 0;
```

Comment:
@mr-rego can you generalise this function, rename the sum variable. This memoize function can be used for any functions not only add.